src/receive_imf/tests.rs

```diff
@@ -2928,20 +2929,27 @@ Reply from different address
 }
 
 #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
-async fn test_long_and_duplicated_filenames() -> Result<()> {
+async fn test_weird_and_duplicated_filenames() -> Result<()> {
     let mut tcm = TestContextManager::new();
     let alice = tcm.alice().await;
     let bob = tcm.bob().await;
 
-    for filename_sent in &[
-        "foo.bar very long file name test baz.tar.gz",
-        "foobarabababababababbababababverylongfilenametestbaz.tar.gz",
-        "fooo...tar.gz",
-        "foo. .tar.gz",
-        "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.tar.gz",
-        "a.tar.gz",
-        "a.tar.gz",
-        "a.a..a.a.a.a.tar.gz",
+    for (filename_sent, expected_ext) in &[
+        ("foo.bar very long file name test baz.tar.gz", "tar.gz"),
+        (
+            "foo.barabababababababbababababverylongfilenametestbaz.tar.gz",
+            "tar.gz",
+        ),
+        ("fooo...tar.gz", "..tar.gz"),
```

Comment:
So, empty "subextensions" also participate in a file extension currently. Probably this is useless, but doesn't break anything. Just a side effect of the current implementation